extensions/azure/storage/AzureBlobStorageClient.h

```diff
@@ -38,8 +39,27 @@ class AzureBlobStorageClient : public BlobStorageClient {
   Azure::Storage::Blobs::Models::UploadBlockBlobResult uploadBlob(const PutAzureBlobStorageParameters& params, gsl::span<const uint8_t> buffer) override;
   std::string getUrl(const PutAzureBlobStorageParameters& params) override;
   bool deleteBlob(const DeleteAzureBlobStorageParameters& params) override;
+  std::unique_ptr<io::InputStream> fetchBlob(const FetchAzureBlobStorageParameters& params) override;
 
  private:
+  class AzureBlobStorageInputStream : public io::InputStream {
+   public:
```

Comment:
Consider moving this to the .cpp file as an implementation detail, possibly in an anonymous namespace. It's only ever exposed through a base class reference.

Reply:
Updated in 81a8a876a579facf06fba6ffd8f5b81f57fbf1f1